feat(database): switch to pg15

yungbender opened the pull request:
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

jdobes reviewed (changes requested):
can you request PG 15 in deploy/clowdapp.yaml for ephemeral?

default is still 12 - https://github.com/RedHatInsights/clowder/blob/master/controllers/cloud.redhat.com/providers/database/localdb.go#L124C5-L124C5

jdobes reviewed (changes requested):
- various components emit:
```
WARNING:psycopg:unknown PostgreSQL timezone: 'Etc/UTC'; will use UTC
```
- instruction from README doesn't work anymore:
```
Switch into database container and run database terminal:
docker exec -it vulnerability-engine-database bash -c "psql -d vulnerability"
```

yungbender commented on docker-compose.yml:
```diff
@@ -3,19 +3,26 @@ version: '3'
+
+  ve_database_init:
+    command: >
+      bash -c "/engine/dbupgrade.sh && /engine/local_init_db.sh"
```
should be good now

yungbender commented:
> * various components emit:
> 
> ```
> WARNING:psycopg:unknown PostgreSQL timezone: 'Etc/UTC'; will use UTC
> ```
> 
> * instruction from README doesn't work anymore:
> 
> ```
> Switch into database container and run database terminal:
> docker exec -it vulnerability-engine-database bash -c "psql -d vulnerability"
> ```

The UTC, is caused by the new image, it sets the timezone to be `Etc/UTC` which is the same as `UTC`

```
vulnerability=# select * from pg_timezone_names where name = 'UTC';
 name | abbrev | utc_offset | is_dst 
------+--------+------------+--------
 UTC  | UTC    | 00:00:00   | f
(1 row)

vulnerability=# select * from pg_timezone_names where name = 'Etc/UTC';
  name   | abbrev | utc_offset | is_dst 
---------+--------+------------+--------
 Etc/UTC | UTC    | 00:00:00   | f
(1 row)
```
The psycopg lib has some problem with parsing this timezone, I have changed it to be only the UTC, inside the docker compose launch (to configure `postgres.conf`).

The README issue works now, I have added `PGUSER` env variable, so that `psql` use it by default.
